src/overlays/actors/ovl_Bg_Relay_Objects/z_bg_relay_objects.c

```diff
@@ -32,23 +43,173 @@ const ActorInit Bg_Relay_Objects_InitVars = {
+                this->actionFunc = func_808A90F4;
+            }
+        } else {
+            Flags_SetSwitch(globalCtx, this->switchFlag);
+            this->actionFunc = func_808A91AC;
+            thisx->posRot.pos.y += 120.0f;
+            D_808A9508 |= 1;
+        }
+    }
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, sp24);
+}
+
+void BgRelayObjects_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgRelayObjects* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    if ((this->dyna.actor.params == WINDMILL_ROTATING_GEAR) && (gSaveContext.cutsceneIndex < 0xFFF0)) {
+        gSaveContext.eventChkInf[6] &= ~0x20;
+    }
+}
+
+void func_808A90F4(BgRelayObjects* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, this->switchFlag)) {
+        if (this->timer != 0) {
+            Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_SLIDE_DOOR_OPEN);
+            if (INV_CONTENT(ITEM_HOOKSHOT) != ITEM_NONE) {
+                this->timer = 120;
+            } else {
+                this->timer = 160;
+            }
+        }
+        if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y + 120.0f, 12.0f)) {
+            this->actionFunc = func_808A91AC;
+        }
+    }
+}
+
+void func_808A91AC(BgRelayObjects* this, GlobalContext* globalCtx) {
+    if (this->unk_169 != 5) {
+        if (this->timer != 0) {
+            this->timer--;
+        }
+        func_8002F994(&this->dyna.actor, this->timer);
+    }
+    if ((this->timer == 0) || (this->unk_169 == globalCtx->roomCtx.curRoom.num)) {
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_SLIDE_DOOR_CLOSE);
+        this->actionFunc = func_808A9234;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Relay_Objects/func_808A90F4.s")
+void func_808A9234(BgRelayObjects* this, GlobalContext* globalCtx) {
+    this->dyna.actor.velocity.y += this->dyna.actor.gravity;
+    if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y, this->dyna.actor.velocity.y)) {
+        func_800AA000(this->dyna.actor.xyzDistFromLinkSq, 180, 20, 100);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_STONE_BOUND);
+        if (this->unk_169 != globalCtx->roomCtx.curRoom.num) {
+            func_800788CC(NA_SE_EN_PO_LAUGH);
+            this->timer = 5;
+            this->actionFunc = func_808A932C;
+            return;
+        }
+        Flags_UnsetSwitch(globalCtx, this->switchFlag);
+        this->dyna.actor.flags &= ~0x10;
+        if (globalCtx->roomCtx.curRoom.num == 4) {
+            gSaveContext.timer1State = 0xF;
+        }
+        this->actionFunc = BgRelayObjects_Noop;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Relay_Objects/func_808A91AC.s")
+void BgRelayObjects_Noop(BgRelayObjects* this, GlobalContext* globalCtx) {
+}
```

Comment:
I prefer DoNothing when describing an action, but we will discuss and standardize things like this at a later time. just mentioning it now

Reply:
yeah I'll change that, I thought Noop was kinda stupid tbh but I saw a lot of actors with it

Reply:
its fine depending on context, in this case its an action where the actor is literally doing nothing